### ISSUE TITLE
[FEATURE] Frontmatter support for intros

### ISSUE THREAD

**Issue by matthieu42morin** [StrangeGirlMurph/obsidian-wikipedia-search#28]
# Description of the Problem

If longer/multiple paragraph {intro} is in frontmatter, it creates problems
Whenever there are:

1) Multiple paragraphs - it creates frontmatter properties on each paragraph (What characters from the templatestring influence this?). Or it straight up doesn't accept it for whatever 2second error reason I am not the Flash enough to read.

2) These characters `;` `:` - they may do some problems: 
IPA pronunciation produces this:

![image](https://github.com/StrangeGirlMurph/obsidian-wikipedia-search/assets/72525251/2ddc7b9a-848e-462c-ba93-847a006c38af)

to become this
 
![image](https://github.com/StrangeGirlMurph/obsidian-wikipedia-search/assets/72525251/68ec12a5-1803-40a0-8630-bfa93ad46c0d)

with `;` at the start

3) [Total order](https://en.wikipedia.org/wiki/Total_order), becomes a bit of a total disorder xD :
![image](https://github.com/StrangeGirlMurph/obsidian-wikipedia-search/assets/72525251/b8572eaf-2a7d-4eb8-a6ff-511527d53ff2)

I use a property `lead`, it has a few advantages, one being that it's not editable outside the frontmatter, which is especially good in this case from wikipedia. So I automatically used `lead: {intro}` with this plugin, it works fine for shorter intros...
but it's a guessing game of how notable this person is to have a huge intro.

# Description of the Solution

## 3 options:
1. Don't use it in the frontmatter :D
2. Limit intro to a single paragraph - can it be universal though, what about info value?
3. Extend the auto-cleaning/parsing 

## This works, it needs to be a string, but it looks bad after removing all break lines / paragraph marks 
![image](https://github.com/StrangeGirlMurph/obsidian-wikipedia-search/assets/72525251/23d17eed-4fd7-4c66-ad66-37fb14c51b39)
![image](https://github.com/StrangeGirlMurph/obsidian-wikipedia-search/assets/72525251/c5c7c0d3-83b5-47a1-994f-3ccfb2493aef)

## Extend the Auto-Cleaning/parsing

In case of multiple paragraph cases I found 2 solutions experimenting:

### 1. Use `<br>` line breaks 

_Banners..._
![image](https://github.com/StrangeGirlMurph/obsidian-wikipedia-search/assets/72525251/541d127f-f4ee-4cc3-9065-cdfebba763ab)

![image](https://github.com/StrangeGirlMurph/obsidian-wikipedia-search/assets/72525251/43b8e9bd-a183-4941-9d60-e0fc0dfcf0a8)

### 2. A remote possibility Use YAML sequence / arrays, could be used to select how much info/paragraphs

i.e. sequence = 
``` yaml
key:
  - string1 = paragraph1
  - string2 = paragraph2  
  - string3 = paragraph3
  
```
  = array =
  
``` yaml
key: ['string1', 'string2', 'string3']
```
the former being automatically rewritten to if you used square brackets in obsidian

=> Has some type error mismatch message, but works

### After 2` <br> <br>`Looks the same as on Wikipedia 
![image](https://github.com/StrangeGirlMurph/obsidian-wikipedia-search/assets/72525251/bb9a70c1-1244-4a1f-8113-cfa36ef3136e)

## Conclusion

If the text is plaintext from wiki and even after cleanup of break lines produces some sort of paragraph marks that look like original wiki that screw up frontmatter then changing those to html `<br>` couldn't hurt as they are supported in md?

# Importance

Intro doesn't work in frontmatter without manual manipulation rn without this fix, depends if anybody uses this stuff.

**Comment by StrangeGirlMurph** [comment]
Hmm... I am against making `<br>` be the default. Yes it's valid markdown syntax but your use case is an edge case and it does make things less readable in source mode and makes editing in live preview more complicated.

Before making this work by changing the code I want to be absolutely sure that this isn't possible to achieve right now. Please check the yaml specification again! There are multiline strings which are okay with line breaks and symbols like ; and : 
The indentation might be just the problem.

![image](https://github.com/StrangeGirlMurph/obsidian-wikipedia-search/assets/62220780/b2b00e9d-4773-416a-ab62-bf2505c7a12e)

**Comment by matthieu42morin** [comment]
Indeed, will do

**Comment by matthieu42morin** [comment]
Hello, sry I took longer, wrote a crazy long text talking about what I tried and doesn't work... unnecessary comment, tried a lot of stuff, removed and changed plugins (like the buggy update-time-on-edit should be banned). In the end the question is how does this plugin and obsidian handle text formatting? Stuff gets formatted before I can do anything with it and I can't get the intro text under one scalar block (whichever folded or literal style I take on later) if there is no possible indentation - it either makes the paragraphs into properties or it errors.

Tldr of it all, made myself a script and now I don't have to manually indent spaces to the paragraph.

try this on Kafka for example,
dependencies: dataview and templater
```
---
tags: type/person role/author
aliases: {title}
lead: |
<%*
let textBlock = `{intro}`;
let lines = textBlock.split('\n');
let indentedText = lines.map(line => '  ' + line).join('\n\n');
tR += indentedText;
%>
banner: {thumbnailUrl} 
source: {url}
created: <% tp.file.creation_date() %>
modified: <% tp.file.last_modified_date() %>
template-type: Wiki_Author
template-version: "1.6"
---

## Who this?

> [!summary] 
> `= this.lead`
> 
>  — _Sourced from wiki:[`= this.aliases`](`= this.source`)_
```

**Comment by matthieu42morin** [comment]
@StrangeGirlMurph But now I cannot get around the fact that if the intro contains a backtick it breaks and I cannot do anything about it... Do you have any tips on how to solve this? I can't escape anything using my script before your plugin replaces the {intro} in templateString as templater executes after it ig, but because the text comes preformatted, it breaks it 100%, for example in some cases, because it makes a new paragraph, strings get ended even without " at the end of the previous paragraph in the frontmatter.
I pushed the problem a bit futher to more edge cases :D , even this is predetermined to break.

**Comment by StrangeGirlMurph** [comment]
I think this works great! I recommend you use this code instead though (much more cleaner and keeps the `\`):
```
lead: |
<%* tR += "  " + String.raw`{intro}`.replaceAll("\n", "\n  ") %>
```
I just make the auto cleanup escape the backticks (plus maybe some of the other markdown syntax) and everything should work perfectly. What other edge cases did you have in mind?

**Comment by matthieu42morin** [comment]
Great, much cleaner! Thanks!

It seems you perhaps fixed some of my issues in your latest commit.

Noticed a discussion page in here, perhaps referencing this could help someone
